mm2src/coins/eth/eth_swap_v2/eth_maker_swap_v2.rs

```diff
@@ -0,0 +1,509 @@
+            }
+        };
+        let data = try_tx_s!(self.prepare_refund_maker_payment_secret_data(args).await);
+
+        self.sign_and_send_transaction(
+            U256::from(ZERO_VALUE),
+            Action::Call(maker_swap_v2_contract),
+            data,
+            U256::from(gas_limit),
+        )
+        .compat()
+        .await
+    }
+
+    pub(crate) async fn spend_maker_payment_v2_impl(
+        &self,
+        args: SpendMakerPaymentArgs<'_, Self>,
+    ) -> Result<SignedEthTx, TransactionErr> {
+        let (token_address, gas_limit) = self
+            .gas_limit_v2
+            .gas_limit(&self.coin_type, EthPaymentType::MakerPayments, PaymentMethod::Spend)
+            .map_err(|e| TransactionErr::Plain(ERRL!("{}", e)))?;
+
+        let maker_swap_v2_contract = self
+            .swap_v2_contracts
+            .as_ref()
+            .map(|contracts| contracts.maker_swap_v2_contract)
+            .ok_or_else(|| TransactionErr::Plain(ERRL!("Expected swap_v2_contracts to be Some, but found None")))?;
+
+        let data = try_tx_s!(self.prepare_spend_maker_payment_data(args, token_address).await);
+
+        self.sign_and_send_transaction(
+            U256::from(ZERO_VALUE),
+            Action::Call(maker_swap_v2_contract),
+            data,
+            U256::from(gas_limit),
+        )
+        .compat()
+        .await
+    }
+
+    pub(crate) async fn wait_for_maker_payment_spend_impl(
+        &self,
+        maker_payment: &SignedEthTx,
+        wait_until: u64,
+    ) -> MmResult<SignedEthTx, WaitForPaymentSpendError> {
+        let decoded = {
+            let func = match self.coin_type {
+                EthCoinType::Eth | EthCoinType::Erc20 { .. } => MAKER_SWAP_V2.function("spendMakerPayment")?,
+                EthCoinType::Nft { .. } => {
+                    return MmError::err(WaitForPaymentSpendError::Internal(
+                        "NFT protocol is not supported for ETH and ERC20 Swaps".to_string(),
+                    ));
+                },
+            };
+            decode_contract_call(func, maker_payment.unsigned().data())?
+        };
+        let maker_swap_v2_contract = self
+            .swap_v2_contracts
+            .as_ref()
+            .map(|contracts| contracts.maker_swap_v2_contract)
+            .ok_or_else(|| {
+                WaitForPaymentSpendError::Internal("Expected swap_v2_contracts to be Some, but found None".to_string())
+            })?;
+        loop {
+            let maker_status = self
+                .payment_status_v2(
+                    maker_swap_v2_contract,
+                    decoded[0].clone(), // id from spendMakerPayment
+                    &MAKER_SWAP_V2,
+                    EthPaymentType::MakerPayments,
+                    2,
+                )
+                .await?;
+            if maker_status == U256::from(MakerPaymentStateV2::TakerSpent as u8) {
+                return Ok(maker_payment.clone());
+            }
```

Comment:
Isn't the purpose of this function waiting for confirmations? or are we gonna use this and then wait for confirmations?

Reply:
oh, I was planing to leave a note about this in this pr, but you mentioned it first.

Lets come back to `wait_for_taker_payment_spend` Taker trait method for a bit.

https://github.com/KomodoPlatform/komodo-defi-framework/blob/a538a027436e8f6387d2dadf38b5b752f0e3e8f4/mm2src/mm2_main/src/lp_swap/taker_swap_v2.rs#L1686-L1693

Well yeah we have doc comment `Wait until taker payment spend is found on-chain` but the function name is confusing, it is not waiting for confirmations.
If we study utxo implementation, then we can see that `wait_for_taker_payment_spend` waits for "output" of taker payment tx which is spent by maker's spend_taker_payment tx. 
Thus, the main purpose of this function so to actually **find** maker's spend taker payment tx, not confirm it. 

I did fix for eth coin in separate branch, opened draft PR so it will be easier to see changes
https://github.com/KomodoPlatform/komodo-defi-framework/pull/2261
I removed `wait_for_maker_payment_spend` function, and renamed trait function `wait_for_taker_payment_spend` to `find_taker_payment_spend_tx`. So it literally tells developer that they need to find taker_payment_spend transaction in implementation for new coin.
And I fixed ethcoin implementation, no it tries to find maker's transaction
https://github.com/KomodoPlatform/komodo-defi-framework/pull/2261/files#diff-2372ecc387777c93c82393588e652382e63341cef14b009732296e0c81352ccbR433

**Note:** it uses taker_approve function as in `sign_and_send_taker_funding_spend` for Eth we return approve tx. It allows us not to do separate decoding for `"ethTakerPayment"` and `"erc20TakerPayment"` transactions to decode swap id value.


Additionally, found out that if we need to find maker's spend taker payment tx, to allow taker to extract maker's secret, then we cant use legacy SwapOPs `extract_secret` in taker state machine, as it uses legacy etomic swap contract for EthCoin
https://github.com/KomodoPlatform/komodo-defi-framework/blob/a538a027436e8f6387d2dadf38b5b752f0e3e8f4/mm2src/mm2_main/src/lp_swap/taker_swap_v2.rs#L2073-L2080

Its in wip, but Im going to provide `extract_secret_v2` for taker swap in this draft pr https://github.com/KomodoPlatform/komodo-defi-framework/pull/2261

---

To sum up, I can remove `wait_for_maker_payment_spend` in this PR or it will be removed in next tpu v2 implementation update

`wait_for_taker_payment_spend` will be renamed, ethcoin impl will be fixed and `extract_secret_v2` function will be provided in next PR

Reply:
Decided to remove `wait_for_maker_payment_spend` right in this PR https://github.com/KomodoPlatform/komodo-defi-framework/pull/2211/commits/6ab7002af93068a2cb548706c979c52cc1713e64